src/main/java/uk/ac/susx/tag/classificationframework/datastructures/ProcessedInstance.java

```diff
@@ -50,7 +51,7 @@
  * Date: 07/08/2013
  * Time: 15:39
  */
-public class ProcessedInstance {
+public class ProcessedInstance implements Serializable {
```

Comment:
Do these classes absolutely need to be java serializable? I'm very reticent to change `ProcessedInstance` or `ClusteredProcessedInstance` since they are used across very many other code repositories.

We currently already serialise instances of documents using JSON, through the Gson library. See for example the [save function in the ModalState class](https://github.com/andehr/tag-classification-framework/blob/aece2465f5fe79d00eb866f51975190bdb56ac65/src/main/java/uk/ac/susx/tag/classificationframework/datastructures/ModelState.java#L178-L225) which saves the inner Instance objects (from trainingDocuments).

Reply:
I mentioned in the comments that i needed to that in order to be able to serialize them 
This task was the hardest cause Qiwei doesnt remember so i had to do it this way after a long discussion with jack trying to figure out the right stuff that i need to serialize and as i was doing that i couldnt serialize these classes so i had to do that in order to be able to create the ser file